conn.go

```diff
@@ -169,7 +177,16 @@ func createConn(nextConn net.Conn, flightHandler flightHandler, handshakeMessage
 	// Handle inbound
 	go c.inboundLoop()
 
-	<-c.handshakeCompleted
+	if c.connectTimeout.Seconds() > 0.0 {
+		select {
+		case <-c.handshakeCompleted:
+		case <-time.After(c.connectTimeout):
+			c.log.Errorf("%s", errConnectTimeout)
```

Comment:
(annoying nit) I would drop the log line. I try to only log when we have things happening async we can't handle.

Reply:
Fixed :+1: